### PR TITLE
Add compat data for input-date

### PR DESCRIPTION
**Pull request by maboa**
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date

### CODE REVIEW

File: html/elements/input/date.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+              },
+              "opera": {
+                "version_added": "11"
+              },
+              "opera_android": {
+                "version_added": "11"
+              },
+              "safari": {
+                "version_added": false,
+                "notes": "It is recognized but there is no UI."
```
Comment (ddbeck) [inline html/elements/input/date.json]: I realize it's from the source table, but I think this note is pretty confusing (to me it implies that it doesn't even show a text box—though I checked it and it does). Can we spell this out a little more, maybe something like "The input type is recognized, but there is no date-specific control."

**Comment by ddbeck** [comment]
Oh I realize now I might've jumped the gun on reviewing this, since there's still activity on the initial input, #2198. Happy to revisit this once there's a known-good input type to use as a baseline. Thanks!

**Comment by wbamberg** [comment]
AFAIK the only outstanding issue on https://github.com/mdn/browser-compat-data/pull/2198 is that this and the other input subfeatures should have a description like:

    "description": "<code>type=\"the-thing-it -is\"</code>",

So with the addition of that, this PR should be fine for you to review @ddbeck .

File: html/elements/input/date.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
+{
+  "html": {
+    "elements": {
+      "input": {
+        "input-date": {
+          "__compat": {
+            "mdn_url": "https://developer.mozilla.org/docs/Web/html/Element/input/date",
```
Comment (ddbeck) [inline html/elements/input/date.json]: Needs a description

**Comment by ddbeck** [comment]
OK, this one is ready for you to look at again, @maboa! Thank you!

**Comment by maboa** [comment]
Hopefully, I've addressed the above :)

**Comment by ddbeck** [comment]
Excellent, thank you @maboa!

**Pull request merged**